src/fonts/castletexturefontdata.pas

```diff
@@ -313,9 +315,22 @@ constructor TTextureFontData.Create(const URL: string;
   FontMgr.Resolution := 0;
   FileName := URIToFilenameSafe(URL);
   if FileName = '' then
-    raise Exception.CreateFmt('Cannot read font from URL "%s". Note that right now only local file URLs are supported', [URL]);
+  begin
+    Cache := Download(URL);
+    try
+      FileName := ApplicationConfig('cache_' + ExtractURIName(URL));
+      StreamSaveToFile(Cache, FileName);
+      FileName := URIDeleteProtocol(FileName);
+      IsCachedFile := true;
```

Comment:
Use `URIToFilenameSafe` to convert URL -> filename, not `URIDeleteProtocol`. Sometimes `URIDeleteProtocol` also works, but the `URIToFilenameSafe` is more reliable for this.

A minor suggestion: maybe add another local variable, like `CacheURL` here. In the code above, the `FileName` variable sometimes holds an URL, sometimes a filename --- I usually found it's more obvious to use separate variables for URLs, separate for filenames. So, I would suggest to write:

```
CacheURL := ApplicationConfig('cache_' + ExtractURIName(URL));
StreamSaveToFile(Cache, CacheURL);
FileName := URIToFilenameSafe(CacheURL);
```